### PR TITLE
Move EnC tests to Features layer

### PR DESCRIPTION
**Pull request by tmat**
Actually eliminates EditorFeatures project dependencies

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi FYI
@davidwengier PTAL

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/EditorFeatures/VisualBasicTest/Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.vbproj [hunk 1]
```diff
@@ -71,6 +72,7 @@
     <Import Include="Xunit" />
   </ItemGroup>
   <ItemGroup>
+    <Folder Include="EditAndContinue\" />
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.vbproj:75]: Can be removed

```suggestion
```

File: src/Features/TestUtilities/EditAndContinue/EditAndContinueTestHelpers.cs [hunk 2]
```diff
@@ -506,6 +506,17 @@ public static MatchingPairs ToMatchingPairs(IEnumerable<KeyValuePair<SyntaxNode,
...
                 }));
         }
+
+        public static void SetDocumentsState(DebuggingSession session, Solution solution, CommittedSolution.DocumentState state)
+        {
+            foreach (var project in solution.Projects)
+            {
+                foreach (var document in project.Documents)
+                {
+                    session.LastCommittedSolution.Test_SetDocumentState(document.Id, state);
```
Comment (sharwell) [inline src/Features/TestUtilities/EditAndContinue/EditAndContinueTestHelpers.cs:516]: As a follow-up, this should go through a TestAccessor (which has analyzer validation that it doesn't get called from production code)

File: src/Features/VisualBasicTest/EditAndContinue/StatementEditingTests.vb [hunk 3]
```diff
@@ -4835,7 +4835,7 @@ End Class"
             edits.VerifySemanticDiagnostics()
         End Sub
 
-        <WpfFact(Skip:="https://github.com/dotnet/roslyn/issues/1212"), WorkItem("https://github.com/dotnet/roslyn/issues/1212")>
+        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/1212"), WorkItem("https://github.com/dotnet/roslyn/issues/1212")>
```
Comment (sharwell) [inline src/Features/VisualBasicTest/EditAndContinue/StatementEditingTests.vb:4838]: ❓ All these tests are skipped. Are we expecting them to work without WpfFact?

**Review by sharwell — APPROVED** [review]
This is one of the cases where I wish we had codecov.io active. It would reveal if we accidentally stopped covering some portion of code due to silent bypass of some implementation due to a missing service (e.g. switching from EditorFeatures composition to Features composition, which might have fewer parts present).

File: src/Features/VisualBasicTest/EditAndContinue/StatementEditingTests.vb (hunk 3 above)
Comment (tmat) [reply src/Features/VisualBasicTest/EditAndContinue/StatementEditingTests.vb:4838]: Yes, there should be no UI dependency.

**Pull request merged**